OpenRA.Game/Graphics/SpriteCache.cs

```diff
@@ -117,18 +117,11 @@ public void LoadReservations(ModData modData)
 						if (loadedFrames != null)
 						{
 							var resolved = new Sprite[loadedFrames.Length];
+							resolvedSprites[token] = resolved;
 							var frames = rs.Frames ?? Enumerable.Range(0, loadedFrames.Length);
 
-							// Premultiplied and non-premultiplied sprites must be cached separately
-							// to cover the case where the same image is requested in both versions.
-							// The premultiplied sprites are stored with an index offset for efficiency
-							// rather than allocating a second dictionary.
-							var di = rs.Premultiplied ? loadedFrames.Length : 0;
 							foreach (var i in frames)
-								resolved[i] = spriteCache.GetOrAdd(i + di,
```

Comment:
you are removing the i + di, are you sure you aren't regressing https://github.com/OpenRA/TiberianDawnHD/issues/28 ?

Reply:
The old version combined the frame index and the premultiplied flag into a single int index. In the new version, we're storing those as distinct elements in the key tuple.

Reply:
ah I see